dsl/src/codegen.rs

```diff
@@ -271,9 +279,85 @@ fn render_expr_code(
       s.push_str("tmp }");
       s
     }
+    Expr::Str(_) => todo!(),
+  }
+}
+
+fn default_value_expr(t: &Type) -> String {
+  match t {
+    Type::Int => "0u64".to_string(),
+    Type::String => "String::new()".to_string(),
+    Type::Void => "()".to_string(),
+    Type::Option(_) => "None".to_string(),
+    Type::Array(inner) => format!("Vec::<{}>::new()", rust_type(inner)),
+    Type::Map(k, v) => format!(
+      "std::collections::HashMap::<{}, {}>::new()",
+      rust_type(k),
+      rust_type(v)
+    ),
+    Type::Struct(name, _) | Type::Custom(name) => format!("{}::default()", pascal_case(name)),
   }
 }
 
+fn render_call_step(
+  ir: &IR,
+  current_fiber: &str,
+  current_func_name: &str,
+  current_func: &Func,
+  target: &FuncRef,
+  args: &Vec<Expr>,
+  ret_to: &StepId,
+  bind: &Option<String>,
+) -> String {
+  let mut s = String::new();
+  let ret_state = variant_name(&[current_fiber, current_func_name, &ret_to.0]);
+  if let Some(callee) = find_func(ir, &target.fiber, &target.func) {
+    s.push_str("      StepResult::Next(vec![\n");
+    s.push_str(&format!("        StackEntry::State(State::{}),\n", ret_state));
+    // Compute bind offset if provided
```

Comment:
My humble take is that we can simplify this, but if you think this is easy to work with, I'll step back.